packages/@aws-cdk/pipelines/lib/pipeline.ts

```diff
@@ -321,9 +317,70 @@ class AssetPublishing extends Construct {
       }
     }
   }
+
+  /**
+   * This role is used by both the CodePipeline build action and related CodeBuild project. Consolidating these two
+   * roles into one, and re-using across all assets, saves significant size of the final synthesized output.
+   * Modeled after the CodePipeline role and 'CodePipelineActionRole' roles.
+   */
+  private generateAssetRole() {
+    const assetRole = new iam.Role(this, 'Role', {
+      roleName: PhysicalName.GENERATE_IF_NEEDED,
+      assumedBy: new iam.CompositePrincipal(new iam.ServicePrincipal('codebuild.amazonaws.com'), new iam.AccountPrincipal(Stack.of(this).account)),
+    });
+
+    // Logging permissions
+    const logGroupArn = Stack.of(this).formatArn({
+      service: 'logs',
+      resource: 'log-group',
+      sep: ':',
+      resourceName: '/aws/codebuild/*',
+    });
+    assetRole.addToPolicy(new iam.PolicyStatement({
+      resources: [logGroupArn],
+      actions: ['logs:CreateLogGroup', 'logs:CreateLogStream', 'logs:PutLogEvents'],
+    }));
```

Comment:
Would be nice to add a bunch of `grant` methods to `codebuild.Project`

Reply:
Any 'normal' `grant` methods would include the specific resource ARN, right? Which is exactly what we're trying to avoid here. 

Unless you're picturing more of a static method, where the 
```ts
class Project extends ProjectBase {
  public static grantRead(grantee: iam.IGrantable, project?: Project) { ... }
}
```